cpp/include/cudf/utilities/span.hpp

```diff
@@ -31,6 +31,8 @@
 
 namespace cudf {
 
+/// A constant of type std::size_t that is used to differentiate std::span of static and dynamic
+/// extent
```

Comment:
```suggestion
/// A constant used to differentiate std::span of static and dynamic extent
```
I don't think we need to mention the type in the comment.